runtime-sdk/src/modules/core/mod.rs

```diff
@@ -839,20 +861,27 @@ impl<Cfg: Config> Module<Cfg> {
         ctx: &C,
         _args: (),
     ) -> Result<types::CallDataPublicKeyQueryResponse, Error> {
-        let key_manager = ctx
-            .key_manager()
-            .ok_or_else(|| Error::InvalidArgument(anyhow!("key manager not available")))?;
-        let epoch = ctx.epoch();
-        let public_key = key_manager
-            .get_public_ephemeral_key(callformat::get_key_pair_id(epoch), epoch)
-            .map_err(|err| match err {
-                keymanager::KeyManagerError::InvalidEpoch(..) => {
-                    Error::InvalidCallFormat(anyhow!("invalid epoch"))
-                }
-                _ => Error::Abort(err.into()),
-            })?;
+        Self::calldata_public_key_common(ctx)
+    }
 
-        Ok(types::CallDataPublicKeyQueryResponse { public_key, epoch })
+    /// Retrieve the public key for encrypting call data (internally exposed call).
+    #[handler(call = "core.CallDataPublicKey", internal)]
+    fn internal_call_calldata_public_key<C: Context>(
+        ctx: &C,
+        _args: (),
+    ) -> Result<types::CallDataPublicKeyQueryResponse, Error> {
+        <C::Runtime as Runtime>::Core::use_tx_gas(CALL_CALLDATA_PUBLIC_KEY_GAS)?;
+        Self::calldata_public_key_common(ctx)
+    }
+
+    /// Retrieve the current epoch.
+    #[handler(call = "core.CurrentEpoch", internal)]
+    fn internal_call_current_epoch<C: Context>(
+        ctx: &C,
+        _args: (),
+    ) -> Result<types::CurrentEpochCallResponse, Error> {
+        <C::Runtime as Runtime>::Core::use_tx_gas(CALL_CURRENT_EPOCH_GAS)?;
+        Ok(types::CurrentEpochCallResponse { epoch: ctx.epoch() })
```

Comment:
I think it would be easier (e.g. for parsing in Solidity) if it just returned an `u64` instead of a map.